shared/src/main/java/com/sap/psr/vulas/shared/json/model/ExemptionBug.java

```diff
@@ -0,0 +1,230 @@
+	
+	private static final String ALL = "*";
+	
+	/** Deprecated configuration prefix. **/
+	public static final String DEPRECATED_CFG_PREFIX = "vulas.report.exceptionExcludeBugs";
+		
+	/** New configuration prefix. **/
+	public static final String CFG_PREFIX = "vulas.report.exemptBug";
+	
+	/** The identifier of a bug, or star (*), which means that the exemption applies to all bugs. */ 
+	private String bugId = null;
+
+	/** The digest of a library or star (*), which means that the exemption applies to all libraries. */
+	private String digest = null;
+
+	private String reason = null;
+
+	/**
+	 * Creates a new exemption, whereby parameters equal to null will be interpreted as star (*).
+	 * 
+	 * @param _bug_id
+	 * @param _digest
+	 * @param _reason
+	 */
+	public ExemptionBug(String _bug_id, String _digest, String _reason) {
+		this.bugId  = (_bug_id==null ? ALL : _bug_id);
+		this.digest = (_digest==null ? ALL : _digest);
+		this.reason = _reason;
+	}
+
+	public String getBugId() { return bugId; }
+
+	public String getDigest() { return digest; }	
+
+	@Override
+	public String getReason() { return reason; }
+
+	@Override
+	public boolean isExempted(VulnerableDependency _vd) {
+		// Bug ID
+		boolean is_exempted = ALL.equals(this.bugId)  || this.bugId.equalsIgnoreCase(_vd.getBug().getBugId());
+		
+		// Archive
+		if(is_exempted) {
+			// All
+			if(ALL.equals(this.digest)) { ; }
+
+			// Digest
+			else if(this.digest.startsWith("dig:")) {
+				is_exempted = is_exempted && this.digest.substring(4).equals(_vd.getDep().getLib().getDigest());
+			}
+			
+			//TODO: Support purl format to exempt findings: https://github.com/package-url/purl-spec
+			else if(this.digest.startsWith("pkg:")) {
+				log.warn("Purl format not yet supported");
+				is_exempted = false;
+			}
+		}
+				 
+		return is_exempted;
+	}
+
+	/**
+	 * Reads all {@link Configuration} settings starting with {@link ExemptionBug#CFG_PREFIX} in order to create {@link ExemptionBug}s.
+	 * Also considers the deprecated settings {@link ExemptionBug#DEPRECATED_CFG_PREFIX} and {@link ExemptionBug#CFG_PREFIX_EXEMPTED_SCOPES} for backward compatibility. 
+	 * 
+	 * @param _cfg
+	 * @return
+	 */
+	public static ExemptionSet readFromConfiguration(Configuration _cfg) {
+		final ExemptionSet exempts = new ExemptionSet();
+		
+		// New format
+		Iterator<String> iter = _cfg.subset(CFG_PREFIX).getKeys();
+		while(iter.hasNext()) {
+			final String k = iter.next();
+			if(!k.equals("")) {
+				final String[] key_elements = k.split("\\.");
+				final int l = key_elements.length; 
+				if(l<2) {
+					log.error("Exemption with key [" + CFG_PREFIX + "." + k + "] has less than 2 elements");
+				}
+				else {
+					exempts.add(new ExemptionBug(StringUtil.join(Arrays.copyOfRange(key_elements, 0, l-1), "."), key_elements[l-1], _cfg.getString(CFG_PREFIX + "." + k)));
+				}
+			}
+		}
+		
+		// Deprecated format
+		final String[] bugs = _cfg.getStringArray(DEPRECATED_CFG_PREFIX);
```

Comment:
At this step old exemptions existing in the db are not kept into account as their key does not contain the prefix vulas. and thus doesn't match DEPRECATED_CFG_PREFIX